src/include/rados/librados.h

```diff
@@ -1453,6 +1453,22 @@ CEPH_RADOS_API int rados_read(rados_ioctx_t io, const char *oid, char *buf,
                               size_t len, uint64_t off);
 
 /**
+ * Compute checksum from object data
+ *
+ * The io context determines the snapshot to checksum, if any was set
+ * by rados_ioctx_snap_set_read().
+ *
+ * @param io the context in which to perform the checksum
+ * @param oid the name of the object to checksum
+ * @param len the number of bytes to checksum
+ * @param off the offset to start checksuming in the object
+ * @param pcrc where to store the checksum result
+ * @return negative error code on failure
+ */
+CEPH_RADOS_API int rados_checksum(rados_ioctx_t io, const char *oid,
+                                  size_t len, uint64_t off, uint32_t *pcrc);
```

Comment:
I think this should have an argument for the checksum to use (crc32c, xxhash, etc.).  Possibly an initialization value (usually -1 for crc32c; not sure about the others).  And the output should be a char *out, size_t size in case it's not a 32-bit hash.

or this method should be called crc32c.  but even then it should still have an initial value instead of assuming -1.